addons/website_sale/controllers/main.py

```diff
@@ -246,28 +240,39 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
         if attrib_list:
             post['attrib'] = attrib_list
 
-        Product = request.env['product.template'].with_context(bin_size=True)
-
-        search_product = Product.search(domain)
+        domain = self._get_search_domain(search, attrib_values)
         website_domain = request.website.website_domain()
-        categs_domain = [('parent_id', '=', False)] + website_domain
-        if search:
-            search_categories = Category.search([('product_tmpl_ids', 'in', search_product.ids)] + website_domain).parents_and_self
-            categs_domain.append(('id', 'in', search_categories.ids))
-        else:
-            search_categories = Category
-        categs = Category.search(categs_domain)
+        categ_domain = expression.AND([domain, website_domain])
+        Product = request.env['product.template'].with_context(bin_size=True)
```

Comment:
Not sure bin_size is still required since images are now stored. To be tested, but the goal was to avoid resizing the images for no reason, which was done on the fly before.